components/dashboard/src/components/UsageView.tsx

```diff
@@ -160,13 +179,61 @@ function UsageView({ attributionId }: UsageViewProps) {
 
     const headerTitle = attributionId.kind === "team" ? "Team Usage" : "Personal Usage";
 
+    const FromLink = forwardRef((arg: any, ref: any) => (
+        <div
+            className="text-blue-500 hover:text-blue-600 dark:text-blue-400 dark:hover:text-blue-500 truncate cursor-pointer"
+            onClick={arg.onClick}
+            ref={ref}
+        >
+            {arg.value}
+        </div>
+    ));
+    const ToLink = forwardRef((arg: any, ref: any) => (
+        <div
+            className="text-blue-500 hover:text-blue-600 dark:text-blue-400 dark:hover:text-blue-500 truncate cursor-pointer"
+            onClick={arg.onClick}
+            ref={ref}
+        >
+            {arg.value}
+        </div>
+    ));
+
     return (
         <>
             <Header
-                title={headerTitle}
-                subtitle={`${new Date(startDateOfBillMonth).toLocaleDateString()} - ${new Date(
-                    endDateOfBillMonth,
-                ).toLocaleDateString()} (updated every 15 minutes).`}
+                title={
+                    <div className="flex items-baseline">
+                        <h1 className="tracking-wide">{headerTitle}</h1>
+                        <h2 className="ml-3">(updated every 15 minutes).</h2>
+                    </div>
+                }
+                subtitle={
+                    <h2 className="tracking-wide flex mt-3">
+                        <h2 className="mr-2">Period: </h2>
+                        {startDateError}
+                        <DatePicker
+                            selected={new Date(startDate)}
+                            onChange={(date) => date && setStartDate(date.getTime())}
+                            selectsStart
+                            startDate={new Date(startDate)}
+                            endDate={new Date(endDate)}
+                            maxDate={new Date(endDate)}
+                            customInput={<FromLink />}
+                            dateFormat={"dd/MM/yyyy"}
+                        />
+                        <div className="mx-2">-</div>
+                        <DatePicker
+                            selected={new Date(endDate)}
+                            onChange={(date) => date && setEndDate(date.getTime())}
+                            selectsEnd
+                            startDate={new Date(startDate)}
+                            endDate={new Date(endDate)}
```

Comment:
**suggestion**: Could we show the last day of a month for each month selection? Currently, for example, when selecting October 2022, the range changes to `01/10/2022 - 01/11/2022`, and would be more accurate if this would change to `01/10/2022 - 31/10/2022`.